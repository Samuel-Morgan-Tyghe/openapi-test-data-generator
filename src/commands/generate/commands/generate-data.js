import { Logger } from '../logger/index.js';
import { DEFAULT_ARRAY_TYPE_TEST_DATA_LENGTH } from '../const/index.js';
import {
  createRandomNumberByRange,
  createRandomNumberByMaxValueStateless,
  createRandomStringByRange,
  getRandomYmd,
  getRandomYmdhhmmss,
  createRandomStringByMaxLengtheStateless,
  createRandomBooleanStateless,
  getRandomYmdStateless,
  getRandomYmdhhmmssStateless
} from '../utils/index.js';

export default class GenerateDataCommand {
  _context;
  _schemaDataMap;
  _schemas;

  setContext(context) {
    this._context = context;
    return this;
  }

  exec(schemaNameList, parsedObject) {
    const context = this._context;
    this._schemas = parsedObject.components.schemas;
    const schemaDataMap = new Map();
    schemaNameList.forEach((s) => {
      const schemaName = s.replaceAll('.yaml', '');
      if (context.ignoreList.indexOf(schemaName) === -1) {
        schemaDataMap.set(
          schemaName,
          this._generateTestDataFromSchema(schemaName, {
            n: context.numberOfArrayTestData,
            stateless: context.stateless,
            exampleSuffix: context.exampleSuffix,
            depthOfRecursion: 0
          })
        );
      } else {
        Logger.warn(
          `[ignore] output: ==> ${context.outputPath}/${schemaName}${context.extension}`
        );
      }
    });
    this._schemaDataMap = schemaDataMap;
  }

  getSchemaDataMap() {
    return this._schemaDataMap;
  }

  /**
   * Create schema sample data
   * @param {*} schemas
   * @param {*} name
   * @param {*} options
   * @returns {Array<object>} sample data
   */
  _generateTestDataFromSchema = (name, options) => {
    const schema = this._schemas[name];
    const numberOfArray = options.n;
    const schemaProperties = schema.properties;
    const resultArray = [];

    if (schema.$ref) {
      const schemaName = schema.$ref.replace('#/components/schemas/', '');
      return this._generateTestDataFromSchema(schemaName, {
        ...options,
        depthOfRecursion: options.depthOfRecursion + 1
      });
    }

    for (let i = 0; i < numberOfArray; i++) {
      // create sample data for a specified number of pieces.
      resultArray.push(
        this._createSampleDataFromProperties(schemaProperties, name, options, i)
      );
    }
    return resultArray;
  };

  _createSampleDataFromProperties = (properties, name, options, i = 0) => {
    const element = {};
    const stateless = options.stateless;
    Object.keys(properties).forEach((key) => {
      const property = properties[key];
      let sampleData;

      // enum type
      if (property.enum && property.enum.length > 0) {
        sampleData = `${property.enum[i % property.enum.length]}`;
      }

      // if exists example data
      const suffix = options.exampleSuffix ? `_${i}` : '';
      sampleData =
        property.example && !(property.enum && property.enum.length > 0) ?
        `${property.example}${suffix}` :
        sampleData;

      // create sample data recursively if the property type is $ref.
      if (property.$ref) {
        sampleData = this._generateTestDataFromSchema(
          property.$ref.replace('#/components/schemas/', ''), {...options, n: 1, depthOfRecursion: options.depthOfRecursion + 1 }
        )[0];
      }

      if (property.type === 'object') {
        sampleData = this._createSampleDataFromProperties(
          property.properties,
          name,
          options,
          i
        );
      }

      // otherwise, create sample data from property attributes
      if (sampleData === undefined) {
        const statelessHashKey = `${name}-${key}-${options.depthOfRecursion}-${i}`;
        sampleData = this.createSampleDataFromPropertyAttributes(
          property,
          stateless,
          options.exampleSuffix,
          statelessHashKey
        );
      }
      element[key] = sampleData;
    });
    return element;
  };

  createSampleDataFromPropertyAttributes = (
    property,
    stateless = false,
    exampleSuffix,
    statelessHashKey = ''
  ) => {
    let value;
    switch (property.type) {
      case 'string':
        if (property.example) {
          value = property.example;
          break;
        }

        if (property.format === 'date-time') {
          // create a random date-time data
          value = stateless ?
            getRandomYmdhhmmssStateless('2000/01/01', statelessHashKey) :
            getRandomYmdhhmmss('2000/01/01');
        } else if (property.format === 'date') {
          // create a random date data
          value = stateless ?
            getRandomYmdStateless('2000/01/01', statelessHashKey) :
            getRandomYmd('2000/01/01');
        } else {
          value = stateless ?
            createRandomStringByMaxLengtheStateless(
              statelessHashKey,
              property.maxLength
            ) :
            createRandomStringByRange(property.minLength, property.maxLength);
        }
        break;
      case 'number':
        // create a random number
        value = stateless ?
          createRandomNumberByMaxValueStateless(
            statelessHashKey,
            property.minimum,
            property.maximum
          ) :
          createRandomNumberByRange(property.minimum, property.maximum);
        break;
      case 'integer':
        // create a random number
        value = stateless ?
          createRandomNumberByMaxValueStateless(
            statelessHashKey,
            property.minimum,
            property.maximum
          ) :
          createRandomNumberByRange(property.minimum, property.maximum);
        break;
      case 'array':
        if (property.items === undefined) {
          value = [];
          break;
        }
        const array = [];
        for (let k = 0; k < DEFAULT_ARRAY_TYPE_TEST_DATA_LENGTH; k++) {
          const suffix = exampleSuffix ? `_${k}` : '';
          if (property.items.type === 'object') {
            array.push(
              this.createSampleDataFromPropertyAttributes(
                property.items.properties,
                stateless,
                `${statelessHashKey}-${k}`
              ) + suffix
            );
          } else {
            array.push(
              this.createSampleDataFromPropertyAttributes(
                property.items,
                stateless,
                `${statelessHashKey}-${k}`
              ) + suffix
            );
          }
        }
        value = array;
        break;
      case 'boolean':
        value = stateless ?
          createRandomBooleanStateless(statelessHashKey) :
          Math.random() > 1 / 2;
        break;
      default:
        // otherwise; {}
        value = {};
    }

    return value;
  };

  _getNextDepthOfRecursion = (depth) => {
    if (!depth) {
      return 1;
    }
    return depth + 1;
  };
}