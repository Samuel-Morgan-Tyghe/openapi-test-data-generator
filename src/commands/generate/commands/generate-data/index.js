import { Logger } from '../../logger/index.js';
import {
  MAX_RECURSION,
  NEED_TO_BE_REPLACE_BY_OPENAI_RESPONSE
} from '../../const/index.js';
import { generateNumberTypeData } from './property-type/number.js';
import { generateStringTypeData } from './property-type/string.js';
import { generateBooleanTypeData } from './property-type/boolean.js';
import { generateArrayTypeData } from './property-type/array.js';
import { Type } from '../../../enum/type.js';
import { fetch } from '../../utils/chatGPT.js';

export default class GenerateDataCommand {
  _context;
  _schemaDataMap;
  _schemas;

  setContext(context) {
    this._context = context;
    return this;
  }

  exec(schemaNameList, parsedObject, openApiKey) {
    const context = this._context;
    this._schemas = parsedObject.components.schemas; // all schemas
    let schemaDataMap = {};

    // Generating all schemas' test data from schema name list
    for (let i = 0; i < schemaNameList.length; i++) {
      let targetSchemaName = schemaNameList[i].replaceAll('.yaml', '');
      if (context.ignoreList.indexOf(targetSchemaName) !== -1) {
        Logger.warn(
          `[ignore] output: ==> ${context.outputPath}/${targetSchemaName}${context.extension}`
        );
        continue;
      }

      if (this._schemas[targetSchemaName].$ref)
        targetSchemaName = schema.$ref.replace('#/components/schemas/', '');

      const data = this._generateNumberOfTargetSchemaDataSpecifiedByOption(
        targetSchemaName, {
          n: context.numberOfData,
          stateless: context.stateless,
          ai: context.ai,
          exampleSuffix: context.exampleSuffix,
          depthOfSchemaRefRecursion: 0,
          avoidAIGenerateList: context.avoidAIGenerateList,
          openApiKey
        }
      );
      schemaDataMap = {
        ...schemaDataMap,
        ... {
          [targetSchemaName]: data
        }
      };
    }
    this._schemaDataMap = schemaDataMap;
  }

  replaceStringPropertyDataByOpenAI = async(apiKey, numberOfData) => {
    if (!apiKey) return;
    const propertySet = new Set(); // these properties will be replaced by openai response value
    for (const schemaName of Object.keys(this._schemaDataMap)) {
      this._findPropertyNamesNeedToBeReplaceByOpenAIResponse(
        this._schemaDataMap[schemaName][0],
        propertySet
      );
    }
    const propertyMap = new Map();
    for (let propertyName of propertySet) {
      propertyMap.set(
        propertyName,
        await fetch(apiKey, propertyName, numberOfData)
      );
    }
    console.log(propertyMap);
    if (propertyMap.size === 0) return;
    const replacedSchemaDataMap = this._replaceByOpenaiData(
      undefined,
      this._schemaDataMap,
      propertyMap,
      0
    );

    this._schemaDataMap = replacedSchemaDataMap;
  };

  getSchemaDataMap() {
    return this._schemaDataMap;
  }

  _replaceByOpenaiData = (key, target, propertyMap, index = 0) => {
    if (Array.isArray(target)) {
      const newArray = [];
      for (let i = 0; i < target.length; i++) {
        newArray.push(
          this._replaceByOpenaiData(key, target[i], propertyMap, i)
        );
      }
      return newArray;
    }

    if (typeof target === 'object') {
      const newObject = {};
      const keys = Object.keys(target);
      for (let i = 0; i < keys.length; i++) {
        const k = keys[i];
        const v = target[k];
        newObject[k] = this._replaceByOpenaiData(k, v, propertyMap, index);
      }
      return newObject;
    }

    if (target === NEED_TO_BE_REPLACE_BY_OPENAI_RESPONSE && key !== undefined) {
      return propertyMap.get(key)[index];
    }
    return target;
  };

  _findPropertyNamesNeedToBeReplaceByOpenAIResponse = (obj, propertySet) => {
    const keys = Object.keys(obj);
    for (let i = 0; i < keys.length; i++) {
      const propertyName = keys[i];
      const propertyValue = obj[propertyName];
      if (propertyValue === NEED_TO_BE_REPLACE_BY_OPENAI_RESPONSE) {
        propertySet.add(propertyName);
        continue;
      }
      if (Array.isArray(propertyValue)) {
        propertyValue.forEach((v) => {
          if (v === NEED_TO_BE_REPLACE_BY_OPENAI_RESPONSE) {
            propertySet.add(propertyName);
          } else if (typeof v === 'object') {
            this._findPropertyNamesNeedToBeReplaceByOpenAIResponse(
              v,
              propertySet
            );
          }
        });
      } else if (typeof propertyValue === 'object') {
        this._findPropertyNamesNeedToBeReplaceByOpenAIResponse(
          propertyValue,
          propertySet
        );
      }
    }
  };

  /**
   * Create schema sample data
   * @param {*} schemas
   * @param {*} targetSchemaName
   * @param {*} options
   * @returns {Array<object>} sample data
   */
  _generateNumberOfTargetSchemaDataSpecifiedByOption = (
    targetSchemaName,
    options,
    depthOfSchemaRefRecursion = 0
  ) => {
    const schema = this._schemas[targetSchemaName];
    const numberOfData = options.n;
    const schemaProperties = schema.properties;
    const resultArray = [];
    if (depthOfSchemaRefRecursion >= MAX_RECURSION) {
      return [];
    }
    // create sample data for a specified number of pieces.
    for (let i = 0; i < numberOfData; i++) {
      resultArray.push(
        this._generateDataFromTargetSchemaProperties(
          schemaProperties,
          targetSchemaName,
          options,
          i,
          depthOfSchemaRefRecursion
        )
      );
    }
    return resultArray;
  };

  _generateDataFromTargetSchemaProperties = (
    properties,
    name,
    options,
    indexOfMultiData = 0,
    depthOfSchemaRefRecursion
  ) => {
    const element = {};
    const stateless = options.stateless;

    if (depthOfSchemaRefRecursion >= MAX_RECURSION) return;

    // Generating data from properties
    Object.keys(properties).forEach((key) => {
      const property = {...properties[key], key };
      let data;
      // enum type
      if (property.enum && property.enum.length > 0) {
        data = `${property.enum[indexOfMultiData % property.enum.length]}`;
      }

      // if exists example and NOT ai mode, data will not be generated randomly but set example with suffix
      const suffix = options.exampleSuffix ? `_${indexOfMultiData}` : '';
      data =
        property.example &&
        !options.ai &&
        !(property.enum && property.enum.length > 0) ?
        `${property.example}${suffix}` :
        data;

      // generate test data recursively if the property type is $ref.
      if (property.$ref) {
        data = this._generateNumberOfTargetSchemaDataSpecifiedByOption(
          property.$ref.replace('#/components/schemas/', ''), {...options, n: 1 },
          depthOfSchemaRefRecursion + 1
        )[0];
      }

      if (property.type === Type.OBJECT) {
        data = this._generateDataFromTargetSchemaProperties(
          property.properties,
          name,
          options,
          indexOfMultiData,
          depthOfSchemaRefRecursion + 1
        );
      }
      // otherwise, generate data from property type
      if (data === undefined) {
        const statelessHashKey = `${name}-${key}-${depthOfSchemaRefRecursion}-${indexOfMultiData}`;
        data = this._generateDataFromPropertyType(
          property,
          stateless,
          options,
          depthOfSchemaRefRecursion,
          statelessHashKey
        );
      }
      element[key] = data;
    });
    return element;
  };

  _generateDataFromPropertyType = (
    property,
    stateless = false,
    options,
    depthOfSchemaRefRecursion,
    statelessHashKey = ''
  ) => {
    let value;
    switch (property.type) {
      case Type.STRING:
        value = generateStringTypeData(
          property,
          statelessHashKey,
          stateless,
          options
        );
        break;
      case Type.NUMBER:
        // create a random number
        value = generateNumberTypeData(property, statelessHashKey, stateless);
        break;
      case Type.INTEGER:
        // create a random number
        value = generateNumberTypeData(property, statelessHashKey, stateless);
        break;
      case Type.BOOLEAN:
        value = generateBooleanTypeData(statelessHashKey, stateless);
        break;
      case Type.ARRAY:
        value = generateArrayTypeData(
          this._schemas,
          property,
          stateless,
          options,
          depthOfSchemaRefRecursion,
          statelessHashKey,
          this._generateDataFromTargetSchemaProperties,
          this._generateDataFromPropertyType
        );
        break;
      default:
        // otherwise; {}
        value = {};
    }

    return value;
  };
}