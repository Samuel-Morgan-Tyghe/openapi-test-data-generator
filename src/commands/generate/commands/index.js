export default class Generator {
  constructor(context, readCommand, generateCommand, outputCommand) {
    this.context = context;
    this._readCommand = readCommand.setContext(context);
    this._generateCommand = generateCommand.setContext(context);
    this._outputCommand = outputCommand.setContext(context);
  }

  async exec() {
    const input = await this.read();
    const parsedObject = await this.generate(input);
    await this.output(parsedObject);
    return parsedObject;
  }

  async read() {
    await this._readCommand.exec();
    const parsedObject = this._readCommand.getParsedObject();
    const schemas = this._readCommand.getSchemas();
    return { parsedObject, schemas };
  }

  async generate(input) {
    this._generateCommand.exec(input.schemas, input.parsedObject);
    return this._generateCommand.getSchemaDataMap();
  }

  async output(parsedObject) {
    await this._outputCommand.exec(parsedObject);
  }
}