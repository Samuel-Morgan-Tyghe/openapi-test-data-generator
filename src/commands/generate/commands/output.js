import { exportFileBySampleData } from '../utils/index.js';

export default class OutputCommand {
  _context;

  setContext(context) {
    this._context = context;
    return this;
  }

  async exec(schemaDataMap) {
    const context = this._context;
    console.log(schemaDataMap);
    for (let key of Object.keys(schemaDataMap)) {
      await exportFileBySampleData(
        key,
        schemaDataMap[key],
        context.outputPath,
        context.extension
      );
    }
  }
}