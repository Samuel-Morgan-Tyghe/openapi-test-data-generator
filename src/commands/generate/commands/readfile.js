import { parse, createSchemas, checkFileExist } from '../utils/index.js';
import { Logger } from '../logger/index.js';
import path from 'path';

export default class ReadFileCommand {
  _context;
  _parsedObject;
  _schemaNameList;

  setContext(context) {
    this._context = context;
    return this;
  }

  async exec() {
    const inputPath = this._context.inputPath;
    const ai = this._context.ai;
    const parsedObject = await parse(path.resolve(process.cwd(), inputPath));

    // read yaml file
    if (parsedObject === null) {
      throw new Error(`Unexpected yaml parse error occured!`);
    }
    const schemaNameList = createSchemas(parsedObject);
    if (schemaNameList === null) {
      throw new Error('Unexpected yaml type. Schemas does not exists.');
    }

    this._schemaNameList = schemaNameList;
    this._parsedObject = parsedObject;
  }

  getParsedObject() {
    return this._parsedObject;
  }

  getSchemaNameList() {
    return this._schemaNameList;
  }
}