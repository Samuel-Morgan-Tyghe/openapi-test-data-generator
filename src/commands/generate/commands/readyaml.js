import { parse, createSchemas } from '../utils/index.js';
import { Logger } from '../logger/index.js';
import path from 'path';

export default class ReadYamlCommand {
	_context;
	_parsedObject;
	_schemaNameList;

	setContext(context) {
		this._context = context;
		return this;
	}

	async exec() {
		const inputPath = this._context.inputPath;
		const parsedObject = await parse(path.resolve(process.cwd(), inputPath));

		if (parsedObject === null) {
			Logger.error(`Unexpected yaml parse error occured!`);
			process.exit(1);
		}
		const schemaNameList = createSchemas(parsedObject);
		if (schemaNameList === null) {
			Logger.error('Unexpected yaml type. Schemas does not exists.');
			process.exit(1);
		}

		this._schemaNameList = schemaNameList;
		this._parsedObject = parsedObject;
	}

	getParsedObject() {
		return this._parsedObject;
	}

	getSchemaNameList() {
		return this._schemaNameList;
	}
}
