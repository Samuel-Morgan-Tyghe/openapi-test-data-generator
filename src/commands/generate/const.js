export const DEFAULT_MAX_RANDOM_VALUE = 1000000