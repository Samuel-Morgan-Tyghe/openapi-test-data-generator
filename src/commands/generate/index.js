import { validate } from './validator/index.js';
import ReadYamlCommand from './commands/readyaml.js';
import GenerateDataCommand from './commands/generate-data.js';
import OutputCommand from './commands/output.js';
import Generator from './commands/index.js';

/**
 * generate test data
 * @param {*} args
 * @returns {Map}
 */
export const generate = async(args) => {
  const context = await validate(args);

  const generator = new Generator(
    context,
    new ReadYamlCommand(),
    new GenerateDataCommand(),
    new OutputCommand()
  );
  await generator.exec();

  return generator.getGeneratedDataObject();
};