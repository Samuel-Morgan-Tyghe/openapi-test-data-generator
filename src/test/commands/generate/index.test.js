import { generate } from '../../../commands/generate';
import fs from 'fs';
import path from 'path';
import { jest } from '@jest/globals';
import { fileURLToPath } from 'url';

jest.mock('fs');
const __filename = fileURLToPath(
  import.meta.url);
const __dirname = path.dirname(__filename);
const outputFilesPath = path.join(__dirname, 'output');

describe('generate-command', () => {
  beforeEach(async() => {
    fs.mkdirSync(outputFilesPath);
  });
  afterEach(async() => {
    fs.rmSync(outputFilesPath, { recursive: true, force: true });
  });
  test('The number of properties', async() => {
    const result = await generate({
      input: `${__dirname}/test-data/openapi-snapshot.yaml`,
      output: outputFilesPath,
      numberOfArrayData: 3,
      extension: '.ts'
    });
    expect(result.size).toBe(3);
    expect(result.has('Employee')).toBe(true);
    expect(result.has('Profile')).toBe(true);
    const employee = result.get('Employee');
    const profile = result.get('Profile');
    const member = result.get('Member');
    expect(employee.length).toBe(3);
    expect(profile.length).toBe(3);
    expect(member.length).toBe(3);
  });

  test('Check if there are correct extension(.ts) files', async() => {
    await generate({
      input: `${__dirname}/test-data/openapi-snapshot.yaml`,
      output: outputFilesPath,
      numberOfArrayData: 3,
      extension: '.ts'
    });
    expect(fs.existsSync(`${outputFilesPath}/Employee.ts`)).toBe(true);
    expect(fs.existsSync(`${outputFilesPath}/Profile.ts`)).toBe(true);
    expect(fs.existsSync(`${outputFilesPath}/Employee.js`)).toBe(false);
    expect(fs.existsSync(`${outputFilesPath}/Profile.js`)).toBe(false);
    expect(fs.existsSync(`${outputFilesPath}/Member.js`)).toBe(false);
    expect(fs.existsSync(`${outputFilesPath}/Member.js`)).toBe(false);
  });
  test('Check if there are correct extension(.js) files', async() => {
    await generate({
      input: `${__dirname}/test-data/openapi-snapshot.yaml`,
      output: outputFilesPath,
      numberOfArrayData: 3,
      extension: '.js'
    });
    expect(fs.existsSync(`${outputFilesPath}/Employee.ts`)).toBe(false);
    expect(fs.existsSync(`${outputFilesPath}/Profile.ts`)).toBe(false);
    expect(fs.existsSync(`${outputFilesPath}/Employee.js`)).toBe(true);
    expect(fs.existsSync(`${outputFilesPath}/Profile.js`)).toBe(true);
    expect(fs.existsSync(`${outputFilesPath}/Member.js`)).toBe(true);
  });

  test('-ignore option', async() => {
    await generate({
      input: `${__dirname}/test-data/openapi-snapshot.yaml`,
      output: outputFilesPath,
      numberOfArrayData: 3,
      ignore: 'Employee'
    });
    expect(fs.existsSync(`${outputFilesPath}/Employee.ts`)).toBe(false);
    expect(fs.existsSync(`${outputFilesPath}/Profile.ts`)).toBe(true);
    expect(fs.existsSync(`${outputFilesPath}/Member.ts`)).toBe(true);
  });

  test('-ignore: multiple ignore', async() => {
    await generate({
      input: `${__dirname}/test-data/openapi-snapshot.yaml`,
      output: outputFilesPath,
      numberOfArrayData: 3,
      ignore: 'Employee, Profile'
    });
    expect(fs.existsSync(`${outputFilesPath}/Employee.ts`)).toBe(false);
    expect(fs.existsSync(`${outputFilesPath}/Profile.ts`)).toBe(false);
    expect(fs.existsSync(`${outputFilesPath}/Member.ts`)).toBe(true);
  });

  test('stateless snapshot test', async() => {
    const result = await generate({
      input: `${__dirname}/test-data/openapi-snapshot.yaml`,
      output: outputFilesPath,
      numberOfArrayData: 4,
      extension: '.ts',
      stateless: 'true'
    });
    expect(result).toMatchSnapshot();
  });
});