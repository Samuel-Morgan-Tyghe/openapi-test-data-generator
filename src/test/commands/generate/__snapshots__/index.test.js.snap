// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate-command stateless snapshot test 1`] = `
Map {
  "Employee" => [
    {
      "companyId": "999_0",
      "department": {
        "id": 4,
        "name": "Marketing",
      },
      "id": "9c0f76ea11",
      "name": "John Doe_0",
      "profile": {
        "age": 40,
        "birthdate": "2026-08-28",
        "hobby": [
          "watching movies_0",
          "watching movies_1",
          "watching movies_2",
        ],
        "sex": "MALE",
      },
      "registdate": "2022-10-03 12:00:00",
      "role": "MANAGER",
    },
    {
      "companyId": "999_1",
      "department": {
        "id": 0,
        "name": "Human Resouce",
      },
      "id": "d1adcc3f59",
      "name": "John Doe_1",
      "profile": {
        "age": 40,
        "birthdate": "2026-08-28",
        "hobby": [
          "watching movies_0",
          "watching movies_1",
          "watching movies_2",
        ],
        "sex": "MALE",
      },
      "registdate": "2029-09-07 08:08:08",
      "role": "ENGINEER",
    },
    {
      "companyId": "999_2",
      "department": {
        "id": 4,
        "name": "Development",
      },
      "id": "a7e94f011e",
      "name": "John Doe_2",
      "profile": {
        "age": 40,
        "birthdate": "2026-08-28",
        "hobby": [
          "watching movies_0",
          "watching movies_1",
          "watching movies_2",
        ],
        "sex": "MALE",
      },
      "registdate": "2011-11-22 11:59:59",
      "role": "ASSISTANT",
    },
  ],
  "Profile" => [
    {
      "age": 46,
      "birthdate": "2011-04-12",
      "hobby": [
        "watching movies_0",
        "watching movies_1",
        "watching movies_2",
      ],
      "sex": "MALE",
    },
    {
      "age": 48,
      "birthdate": "2017-12-22",
      "hobby": [
        "watching movies_0",
        "watching movies_1",
        "watching movies_2",
      ],
      "sex": "FEMALE",
    },
    {
      "age": 32,
      "birthdate": "2024-04-09",
      "hobby": [
        "watching movies_0",
        "watching movies_1",
        "watching movies_2",
      ],
      "sex": "ELSE",
    },
  ],
}
`;
